From 7ea549a75a4e593863dc03bb316c5fb53796be8c Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg@freescale.com>
Date: Wed, 2 Jan 2013 15:10:41 -0600
Subject: [PATCH 136/281] ENGR00238582-1: iMX53 SMD change for jellybean 4.2

Some more changes for getting Jellybean 4.2 running
on iMX53 SMD.

Signed-off-by: Nitin Garg <nitin.garg@freescale.com>
---
 imx53_smd/init.freescale.rc | 3 +++
 imx5x/imx5x.mk              | 2 +-
 imx5x/init.rc               | 7 ++++++-
 imx5x/ueventd.freescale.rc  | 4 ++--
 4 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/imx53_smd/init.freescale.rc b/imx53_smd/init.freescale.rc
index 2fbbcc9..32fbb76 100755
--- a/imx53_smd/init.freescale.rc
+++ b/imx53_smd/init.freescale.rc
@@ -1,5 +1,8 @@
 import init.${ro.hardware}.usb.rc
 
+on init
+    start watchdogd
+
 on boot
 
     # Set permission for IIM node
diff --git a/imx5x/imx5x.mk b/imx5x/imx5x.mk
index 72d169e..dd7bd4a 100644
--- a/imx5x/imx5x.mk
+++ b/imx5x/imx5x.mk
@@ -1,4 +1,4 @@
-$(call inherit-product, $(SRC_TARGET_DIR)/product/full_base_telephony.mk)
+$(call inherit-product, $(SRC_TARGET_DIR)/product/full_base.mk)
 
 # overrides
 PRODUCT_BRAND := Freescale
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 5943568..ef880b0 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -38,7 +38,7 @@ loglevel 3
     export LOOP_MOUNTPOINT /mnt/obb
     export SHM_MOUNTPOINT /mnt/shm
     export EXTERNAL_STORAGE /mnt/sdcard
-    export BOOTCLASSPATH /system/framework/core.jar:/system/framework/core-junit.jar:/system/framework/bouncycastle.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/android.policy.jar:/system/framework/services.jar:/system/framework/apache-xml.jar
+    export BOOTCLASSPATH /system/framework/core.jar:/system/framework/core-junit.jar:/system/framework/bouncycastle.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/telephony-common.jar:/system/framework/mms-common.jar:/system/framework/android.policy.jar:/system/framework/services.jar:/system/framework/apache-xml.jar
 
     export LIGHTSENSOR_PATH /sys/class/i2c-adapter/i2c-1/1-0044
     export VPU_FW_PATH /system/lib/firmware/vpu
@@ -204,6 +204,8 @@ on post-fs-data
 
     write /proc/apanic_console 1
 
+    mkdir /data/media 0770 media_rw media_rw
+
     # create basic filesystem structure
     mkdir /data/misc 01771 system misc
     mkdir /data/misc/adb 02750 system shell
@@ -623,3 +625,6 @@ service bugreport /system/bin/bugmailer.sh -b /system/media/audio/ui/Dock.ogg -e
     oneshot
     keycodes 115 102
 
+# Set watchdog timer to 30 seconds and pet it every 10 seconds to get a 20 second margin
+service watchdogd /sbin/watchdogd 10 20
+    class core
diff --git a/imx5x/ueventd.freescale.rc b/imx5x/ueventd.freescale.rc
index b5f0771..a528024 100644
--- a/imx5x/ueventd.freescale.rc
+++ b/imx5x/ueventd.freescale.rc
@@ -1,6 +1,4 @@
-/dev/ttymxc1              0600   bluetooth  bluetooth
 /dev/ttymxc2              0600   bluetooth  bluetooth
-/dev/ion                  0666   media      system
 /dev/snd/*                0664   system     audio
 /dev/ttyUSB*              0640   radio      radio
 /dev/ttyACM*              0640   radio      radio
@@ -19,7 +17,9 @@
 /dev/ptya0                0660   system     gps
 /dev/ttya1                0660   system     gps
 /dev/ptya1                0660   system     gps
+/dev/ion                  0666   media      system
 /dev/graphics/fb0         0660   system     graphics
+/dev/watchdog             0660   system     root
 
 # sysfs properties
 /sys/devices/virtual/input/input*   name        0660  root   input
-- 
1.8.0

